operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java

```diff
@@ -0,0 +1,140 @@
+import io.quarkiverse.operatorsdk.runtime.OperatorProducer;
+import io.quarkiverse.operatorsdk.runtime.QuarkusConfigurationService;
+import io.quarkus.logging.Log;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.util.TypeLiteral;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.List;
+import java.util.UUID;
+
+public abstract class ClusterOperatorTest {
+
+  public static final String QUARKUS_KUBERNETES_DEPLOYMENT_TARGET = "quarkus.kubernetes.deployment-target";
+  public static final String OPERATOR_DEPLOYMENT_PROP = "test.operator.deployment";
+  public static final String TARGET_KUBERNETES_GENERATED_YML_FOLDER = "target/kubernetes/";
+
+  public enum OperatorDeployment {local,remote}
+
+  protected static OperatorDeployment operatorDeployment;
+  protected static Instance<Reconciler<? extends HasMetadata>> reconcilers;
+  protected static QuarkusConfigurationService configuration;
+  protected static KubernetesClient k8sclient;
+  protected static String namespace;
+  protected static String deploymentTarget;
+  private static Operator operator;
+
+
+  @BeforeAll
+  public static void before() throws FileNotFoundException {
+    configuration = CDI.current().select(QuarkusConfigurationService.class).get();
+    reconcilers = CDI.current().select(new TypeLiteral<>() {});
+    operatorDeployment = ConfigProvider.getConfig().getOptionalValue(OPERATOR_DEPLOYMENT_PROP, OperatorDeployment.class).orElse(OperatorDeployment.local);
+    deploymentTarget = ConfigProvider.getConfig().getOptionalValue(QUARKUS_KUBERNETES_DEPLOYMENT_TARGET, String.class).orElse("kubernetes");
+
+    calculateNamespace();
+    createK8sClient();
+    createNamespace();
+
+    if (operatorDeployment == OperatorDeployment.remote) {
+      createCRD();
+      createRBACresourcesAndOperatorDeployment();
+    } else {
+      createOperator();
+      registerReconcilers();
+      operator.start();
+    }
+
+  }
+
+  private static void createK8sClient() {
+    k8sclient = new DefaultKubernetesClient(new ConfigBuilder(Config.autoConfigure(null)).withNamespace(namespace).build());
+  }
+
+  private static void createRBACresourcesAndOperatorDeployment() throws FileNotFoundException {
+    Log.info("Creating RBAC into Namespace " + namespace);
+    List<HasMetadata> hasMetadata = k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + deploymentTarget + ".yml"))
+            .inNamespace(namespace).get();
+    hasMetadata.stream()
+            .map(b -> {
+              if ("Deployment".equalsIgnoreCase(b.getKind()) && b.getMetadata().getName().contains("operator")) {
+                ((Deployment) b).getSpec().getTemplate().getSpec().getContainers().get(0).setImagePullPolicy("Never");
```

Comment:
This is not needed, cloning this branch and:
```
mvn clean package
```
the Deployment already has: `imagePullPolicy: Never`.
Also this can prevent from running on an external Kubernetes (e.g. non minikube).

Reply:
1. this will ensure that test always has Never.... as the global configuration we have with Never should change in the future to allow deploying in non test environments
2. I dont get why this prevent on running in external kubernetes because this is ensuring that in TEST this Never is the value to use.

Reply:
1. `minikube` is not going to be our only test environment, and this will make it fail anywhere else
2. in `TEST` we should be able to run the deployment in e.g. an OpenShift external cluster, this check and substitution will prevent is.

this is not even needed ATM, I'm convinced that we should remove it (at least for now).

Reply:
This obviously can't work in our QE env, but it's ok for the prototype. We can't hold this PR due to discussions for much longer...

Reply:
> We can't hold this PR due to discussions for much longer...

I agree that we should not keep this PR open much longer, but I don't think this comment is the cause of any additional harm or delay.